AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m

```diff
@@ -1,23 +1,37 @@
 #import "MSAnalyticsInternal.h"
 #import "MSAnalyticsTransmissionTargetInternal.h"
 #import "MSAnalyticsTransmissionTargetPrivate.h"
+#import "MSMockUserDefaults.h"
 #import "MSTestFrameworks.h"
 
 static NSString *const kMSTestTransmissionToken = @"TestTransmissionToken";
 static NSString *const kMSTestTransmissionToken2 = @"TestTransmissionToken2";
 
 @interface MSAnalyticsTransmissionTargetTests : XCTestCase
+
+@property(nonatomic) MSMockUserDefaults *storageMock;
 @end
 
 @implementation MSAnalyticsTransmissionTargetTests
 
+- (void)setUp {
+  [super setUp];
+  self.storageMock = [MSMockUserDefaults new];
+
+  // Analytics enabled state can prevent targets from tracking events.
+  id AnalyticsClassMock = OCMClassMock([MSAnalytics class]);
+  OCMStub(ClassMethod([AnalyticsClassMock isEnabled])).andReturn(YES);
+}
+
 #pragma mark - Tests
 
 - (void)testInitialization {
 
   // When
   MSAnalyticsTransmissionTarget *transmissionTarget =
-      [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:kMSTestTransmissionToken];
+      [[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:kMSTestTransmissionToken
+                                                                parentTarget:nil
+                                                                     storage:self.storageMock];
```

Comment:
Probably this is why you added an init with storage parameter in private header. We have settings mock which also mocks `NSUserDefaults` and you can use this one instead.

Reply:
We can use `MSMockUserDefault` in this case as discussed offline. Let's add a TODO for this one if you don't have time to address.

Reply:
Yes `MSMockUserDefault` will replace `NSUserDefaults` instance at run time, so no need to pass it around. will need to clean the AbstractService class/tests too. TODO added.